### PR TITLE
feat(lib): add input to control select on click

### PR DESCRIPTION
**Pull request by sam-eah**
This small PR gives the ability to chose wether an item should be selected by clicking on it. This is done using a new input,  `selectOnClick` in `dts-select-container`.  
I'm using this library with some custom drag & drop for a file explorer project. I need to be able to drag & drop all selected items together, however at the current state it is not possible, since starting to drag by clicking on an item will unselect all other items. This fixes this issue.  

Thanks again for the awesome work :)

### CODE REVIEW

**Comment by d3lm** [comment]
Hey! Thanks for the PR. I am not sure I understand cause you can use the `selectMode`, see https://github.com/d3lm/ngx-drag-to-select#dts-select-container. Have you tried to enable that?

**Comment by d3lm** [comment]
Strange. I just looked at the linting error and it's a little weird.

**Comment by sam-eah** [comment]
> Do you simply want to disable selection on click but still be able to drag to select items?  

Yes, exactly !

**Comment by d3lm** [comment]
Makes sense. I am down to continue with this PR if you add tests and fix CI.

**Comment by sam-eah** [comment]
It's a bit strange, I juste tried the cypress tests on my local server and they all passed 
e2e
And I got no errors from linting or styling with the commands `style` and `lint:check` either  
Do you have any idea ?

**Comment by d3lm** [comment]
Hm that's indeed very strange. I can look into this on the weekend if I find the time and give it a try on my machine.

**Comment by d3lm** [comment]
I have also restarted all jobs. Maybe that helps 🤷‍♂️

**Comment by sam-eah** [comment]
> I have also restarted all jobs. Maybe that helps 🤷‍♂️

Okay now all Cypress tests pass!  
But the lint task still fails: `npm ERR! code ELIFECYCLE`
I would typically try these commands for this error, but I'm not sure how github CI pipelines handle node modules 😓 
- `npm cache clean --force`
- `rm -rf node_modules package-lock.json`
- `npm install`

**Comment by sam-eah** [comment]
Hello, any updates ?

**Comment by d3lm** [comment]
Sorry for my late response. Been super busy lately. I try to have a look this week to see if I can find out why the Lint stage fails and how I can fix it. It also fails in another PR 🤔

**Comment by sam-eah** [comment]
No worries, thanks for your time!

File: README.md [hunk 1]
```diff
@@ -250,6 +250,7 @@ shortcuts: {
 | --------------------- | ---------- | ------- | ------------------------------------------------------------------------------------------------------------- |
 | selectedItems         | Array<any> | /       | Collection of items that are currently selected                                                               |
 | selectOnDrag          | Boolean    | `true`  | Whether items should be selected while dragging                                                               |
+| selectOnClick         | Boolean    | `true`  | Whether items should be selected when clicking them                                                           |
```
Comment (d3lm) [inline README.md:253]:
```suggestion
| selectOnClick         | Boolean    | `true`  | Whether items should be selected when clicked                                                           |
```

**Review by d3lm — CHANGES_REQUESTED** [review]
I reviewed your PR and have a small nitpick. Also, please add e2e tests for this feature.

**Comment by d3lm** [comment]
I just merged another PR. Please rebase this PR again on latest master. Thanks!

**Review by d3lm — CHANGES_REQUESTED** [review]
Ok this looks good, but can you please add some e2e tests for this?

**Comment by sam-eah** [comment]
Hello, I've added some e2e tests, please tell me if this suits you

File: cypress/integration/clicking.spec.ts [hunk 2]
```diff
@@ -190,6 +191,18 @@ describe('Clicking', () => {
     });
   });
 
+  it('should not select a single item when selectOnClick is false', () => {
+    getDesktopExample().within(() => {
+      disableSelectOnClick();
```
Comment (d3lm) [inline cypress/integration/clicking.spec.ts:196]: Can you add a newline after this?

**Review by d3lm — CHANGES_REQUESTED** [review]
Yep this looks good. Thanks for adding a test. Left one minor comment.

File: cypress/integration/clicking.spec.ts (hunk 2 above)
Comment (sam-eah) [reply cypress/integration/clicking.spec.ts:196]:
```suggestion
      disableSelectOnClick();
      
```
Like this?

Comment (d3lm) [reply cypress/integration/clicking.spec.ts:196]: Yup 👍

**Review by d3lm — APPROVED** [review]

**Comment by sam-eah** [comment]
Awesome! Do you know when a new npm version can be released?

**Comment by d3lm** [comment]
It's now published. You can grab 4.2.0.

**Comment by sam-eah** [comment]
Amazing, thanks!

**Pull request merged**